Firestore/core/src/local/leveldb_document_overlay_cache.cc

```diff
@@ -258,6 +250,40 @@ void LevelDbDocumentOverlayCache::ForEachKeyWithLargestBatchId(
   }
 }
 
+void LevelDbDocumentOverlayCache::ForEachKeyInCollection(
+    const ResourcePath& collection,
+    int since_batch_id,
+    std::function<void(LevelDbDocumentOverlayKey&&)> callback) const {
+  const std::string index_start_key =
+      LevelDbDocumentOverlayCollectionIndexKey::KeyPrefix(user_id_, collection,
+                                                          since_batch_id + 1);
+  const std::string index_key_prefix =
+      LevelDbDocumentOverlayCollectionIndexKey::KeyPrefix(user_id_, collection);
+
+  auto it = db_->current_transaction()->NewIterator();
+  for (it->Seek(index_start_key);
+       it->Valid() && absl::StartsWith(it->key(), index_key_prefix);
+       it->Next()) {
+    LevelDbDocumentOverlayCollectionIndexKey key;
+    HARD_ASSERT(key.Decode(it->key()));
+    if (key.collection() != collection) {
+      break;
+    }
+    callback(std::move(key).ToLevelDbDocumentOverlayKey());
+  }
+}
+
+absl::optional<Overlay> LevelDbDocumentOverlayCache::GetOverlay(
+    const LevelDbDocumentOverlayKey& key) const {
+  auto it = db_->current_transaction()->NewIterator();
+  const std::string encoded_key = key.Encode();
+  it->Seek(encoded_key);
+  if (!(it->Valid() && it->key() == encoded_key)) {
```

Comment:
I think this would be easier to read if this was `!it->valid() || it->key != encoded_key`. I just spent a couple seconds wondering if the statement was wrong. You can keep as is if you prefer as this is might just be my brains slowness this morning.

Reply:
Done. If you find it confusing, others probably do too. I also made a similar change to two other "if" statements in this file.